http-client/src/main/java/io/micronaut/http/client/DefaultHttpClient.java

```diff
@@ -1796,18 +1796,43 @@ private String getHostHeader(URI requestURI) {
 
             @Override
             protected void channelRead0(ChannelHandlerContext channelHandlerContext, FullHttpResponse fullResponse) {
-                HttpResponseStatus status = fullResponse.status();
-                int statusCode = status.code();
-                HttpStatus httpStatus;
                 try {
-                    httpStatus = HttpStatus.valueOf(statusCode);
-                } catch (IllegalArgumentException e) {
-                    emitter.onError(e);
-                    return;
+
+                    HttpResponseStatus status = fullResponse.status();
+                    int statusCode = status.code();
+                    HttpStatus httpStatus;
+                    try {
+                        httpStatus = HttpStatus.valueOf(statusCode);
+                    } catch (IllegalArgumentException e) {
+                        if (complete.compareAndSet(false, true)) {
+                            emitter.onError(e);
+                        } else {
+                            getLog().warn("unsupported http status after completion", e);
```

Comment:
This isn't how we do logging in the rest of the file. Please change to 
`if (LOG.isWarnEnabled()) { LOG.warn(...) }`

Reply:
Done.